### ISSUE TITLE
Reduce number of attrs for `#[visited_node]` used by AST codegen

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#4281]
Currently:

```rs
#[visited_node]
pub enum Expression<'a> {
    #[visit_args(flags = ScopeFlags::Function)]
    FunctionExpression(Box<'a, Function<'a>>),
    /* ... */
}

#[visited_node]
pub struct TryStatement<'a> {
    /* ... */
    #[visit_as(FinallyClause)]
    pub finalizer: Option<Box<'a, BlockStatement<'a>>>,
}
```

Could we reduce the number of "special" attrs by changing `visit_args(...)` / `visit_as(...)` to `visit(args(...))`  / `visit(as(...))`?

```rs
#[visited_node]
pub enum Expression<'a> {
    #[visit(args(flags = ScopeFlags::Function))]
    FunctionExpression(Box<'a, Function<'a>>),
    /* ... */
}

#[visited_node]
pub struct TryStatement<'a> {
    /* ... */
    #[visit(as(FinallyClause))]
    pub finalizer: Option<Box<'a, BlockStatement<'a>>>,
}
```

@rzvxa What do you think? Not a big deal, but I think it'd be neater. I suspect we may end up adding more attrs later on (it may be useful, for instance, to tag all TS-only fields `#[typescript]`) and it could get out of hand.

**Comment by rzvxa** [comment]
Good idea, I added the `visit_args`, and when the need for `visit_as` was raised I just copied the parsing of the previous one. I'll merge them together.

**Comment by overlookmotel** [comment]
Implemented in #4371.
